File: Java/Easy/Valid-Parentheses.java
class Solution {
    public boolean isValid(String s) {
	Stack<Character> stack = new Stack<Character>();
	for (char c : s.toCharArray()) {
        if (c == '(' || c == '{' || c == '[') {
            stack.push(c);
        } else {
            if (stack.isEmpty() || !checkPair(stack.pop(),c)) {
                return false;
            }
        }
	}
	return stack.isEmpty();
    }
    public boolean checkPair(char popped, char c) {
        return (popped=='(' && c==')' || popped=='[' && c==']' || popped=='{' && c=='}');
    }
}

File: Python/Easy/Valid-Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            if c in ('(','{','['):
                stack.append(c)
            else:
                if len(stack)==0 or not self.checkPair(stack.pop(),c):
                    return False
        return len(stack)==0
    def checkPair(self, popped, c):
        return (popped=='(' and c==')' or popped=='[' and c==']' or popped=='{' and c=='}')
